code/obj/item/beartrap.dm

```diff
@@ -58,6 +58,8 @@
 
 	Crossed(mob/living/M)
 		if (src.armed && istype(M) && !(isintangible(M) || isghostcritter(M)) && !M.isFlying)
+			if (check_target_immunity(target=M, ignore_everything_but_nodamage=FALSE, source=src))
```

Comment:
Not sure source=src is correct.

Reply:
Source currently matters only for projectiles it seems but passing it could be useful for future refinement of the proc.